src/lib/intent-list/index.ts

```diff
@@ -107,26 +109,35 @@ class IntentList {
           const entityType = action.getEntityType()
           const entityId = action.getEntityId()
 
-          if (entityType === 'CONTENT_TYPE') {
+          if (entityType === EntityType.ContentType) {
             const ct = await api.getContentType(entityId)
             await action.applyTo(ct)
           }
-          if (entityType === 'EDITOR_INTERFACE') {
+          if (entityType === EntityType.EditorInterface) {
             try {
               const editorInterfaces = await api.getEditorInterfaces(entityId)
               await action.applyTo(editorInterfaces)
             } catch {
               // TODO: maybe a better handling
             }
           }
+          if (entityType === EntityType.Tag) {
+            const tag = await api.getTag(entityId)
+            await action.applyTo(tag)
+          }
           continue
         }
       }
 
       // Auto insert publish and save
 
       if (intent.shouldSave()) {
-        const save = new ContentTypeSaveAction(intent.getContentTypeId())
+        let save: any
+        if (intent.isTagIntent()) {
+          save = new TagSaveAction(intent.getTagId())
+        } else {
+          save = new ContentTypeSaveAction(intent.getContentTypeId())
+        }
```

Comment:
Another hint that we need to rethink design around intents [as mentioned by Tim](https://github.com/contentful/contentful-migration/pull/416/files#r451394607). Ideally we would have an `intentType` getter or alike, then we could say

```typescript
switch (intent.type) {
  [Intent.Tag]:
    return new TagSaveAction(intent.getTagId())
  [Intent.ContentType]:
  [Intent.EditorInterface]:
    return ContentTypeSaveAction(intent.getContentTypeId())
}
```